hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/qjournal/server/Journal.java

```diff
@@ -264,9 +264,9 @@ void format(NamespaceInfo nsInfo, boolean force) throws IOException {
    */
   @Override // Closeable
   public void close() throws IOException {
-    storage.close();
```

Comment:
This is because it can throw IOException?

Reply:
Yes, exactly. If it throws an exception, then committedTxnId and curSegment are never closed.